src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -800,4 +802,47 @@ function deletesorted!(a::AbstractVector, x)
   found
 end
 
+""" More convenient syntax for declaring an ACSet
+
+Example:
+@acset Graph begin
+  V = 2
+  E = 2
+  src = [1,2]
+  tgt = [2,1]
+end
+"""
+
+macro acset(head, body)
+  expr = :(init_acset($(esc(head)), $(Expr(:quote, body))))
+  Expr(:call, esc(:eval), expr)
+end
+
+"""
+TODO: Well-formedness check
+TODO: Could also rely on a @generated function that took in a "flat" named tuple
+TODO: Alternative syntax for @acset input based on CSV
+TODO: Actual CSV input
+"""
+function init_acset(T::Type{<:ACSet{CD,AD,Ts}},body) where {CD <: CatDesc, AD <: AttrDesc{CD}, Ts <: Tuple}
```

Comment:
This function is publicly exported, but should it be? If it is exported, it should have a real docstring.